raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -405,10 +516,9 @@ def state_transition(
             state_change,
             channelidentifiers_to_channels,
             pseudo_random_generator,
+            block_number,
         )
     else:
         iteration = TransitionResult(payment_state, list())
 
-    sanity_check(iteration.new_state)
```

Comment:
please, don't remove the sanity check

Reply:
The sanity check in this case was replaced by `clear_if_finalized`.
```
def sanity_check(payment_state: InitiatorPaymentState):	
    assert (	
        payment_state is None or payment_state.initiator is not None	
    ), 'either the task must be finished or there must be an initiator transfer pending'
```
Which the new function handles by removing the payment state iff the are no more pending transfers